### PR TITLE
Add Kitodo-script action generateImages

### PR DESCRIPTION
**Pull request by matthias-ronge**
Adds a `generateImages` function to the KitodoScript dialog. It takes two parameters: `folders` and `images`. `folders` can be either `all` or an enumeration (with a comma) of configured folders. `images` can be `all`, `missing` or `missingOrDamaged`.

### CODE REVIEW

File: Kitodo/src/main/java/org/kitodo/production/services/command/KitodoScriptService.java [hunk 1]
```diff
@@ -195,6 +213,54 @@ private void deleteMetadataDirectory(Process process) throws IOException {
...
+            }
+            Subfolder sourceFolder = new Subfolder(process, generatorSource);
+            if (sourceFolder.listContents().isEmpty()) {
+                Helper.setErrorMessage("kitodoScript.generateImages.error.noSourceFiles",
+                    new String[] {process.getTitle(), sourceFolder.getRelativeDirectoryPath() });
+                continue;
+            }
+            List<String> ungeneratableFolders = new ArrayList<>(folders);
+            boolean all = folders.size() == 1 && folders.get(0).equalsIgnoreCase("all");
+            List<Subfolder> outputs = new ArrayList<>();
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/production/services/command/KitodoScriptService.java]:
```suggestion
            List<Subfolder> outputFolders = new ArrayList<>();
```

File: Kitodo/src/main/java/org/kitodo/production/services/command/KitodoScriptService.java [hunk 2]
```diff
@@ -195,6 +213,54 @@ private void deleteMetadataDirectory(Process process) throws IOException {
...
+            if (outputs.isEmpty()) {
+                Helper.setErrorMessage("kitodoScript.generateImages.error.noDestination",
+                    new String[] {process.getTitle(), String.join(", ", ungeneratableFolders) });
+                continue;
+            }
+            ImageGenerator imageGenerator = new ImageGenerator(sourceFolder, generationMode, outputs);
+            TaskManager.addTask(new TaskImageGeneratorThread(process.getTitle(), imageGenerator));
+            String generationModeTranslated = Helper
+                    .getTranslation("imageGenerator.generationMode.".concat(generationMode.toString()));
+            String generatableFolders = outputs.stream().map(Subfolder::getFolder).map(Folder::getPath)
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/production/services/command/KitodoScriptService.java]: actually the folders are not generated, right? They are filled with images?

**Comment by Kathrin-Huber** [comment]
and please add a test ;-)

File: Kitodo/src/main/java/org/kitodo/production/services/command/KitodoScriptService.java (hunk 2 above)
Comment (matthias-ronge) [reply Kitodo/src/main/java/org/kitodo/production/services/command/KitodoScriptService.java]: Folders are never _generated_ (they can be _created_, but that's not the topic here), only the contents in the folders are ever generated.

**Comment by matthias-ronge** [comment]
Apparently, the changes have broken the code. 😞

File: Kitodo/src/test/resources/metadata/2/jpgs/max/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+*
```
Comment (Kathrin-Huber) [inline Kitodo/src/test/resources/metadata/2/jpgs/max/.gitignore]: please remove after test instead of ignoring it.

Comment (matthias-ronge) [reply Kitodo/src/test/resources/metadata/2/jpgs/max/.gitignore]: This is to commit an „empty folder” to git.

**Comment by Kathrin-Huber** [comment]
please resolve conflicts

**Pull request merged**